src/main/java/io/jenkins/plugins/oidc_provider/IdTokenCredentials.java

```diff
@@ -154,12 +168,26 @@ RSAPublicKey publicKey() {
             setAudience(audience).
             setExpiration(Date.from(new Date().toInstant().plus(1, ChronoUnit.HOURS))).
             setIssuedAt(new Date());
+
         if (build != null) {
             builder.setSubject(build.getParent().getAbsoluteUrl()).
                 claim("build_number", build.getNumber());
         } else {
             builder.setSubject(Jenkins.get().getRootUrl());
         }
+
+        // Set claims from token macros, if available
+        HashMap<String, String> tokenMacros = new HashMap<>();
+        tokenMacros.put("git_branch", "${GIT_BRANCH,fullName=false}");
```

Comment:
To make this more general I would suggest checking https://github.com/jenkinsci/branch-api-plugin/blob/7332a9019a0109975bfa537fa00a6f98e74e3721/src/main/resources/jenkins/branch/BranchNameContributor/buildEnv.properties#L23-L37

Reply:
you mean modify the name to branch_name?
I tried to stick with the same token macro names the plugin itself returns, but I'm fine changing them if needed.

Reply:
All of the above bits of information would just be exposed to the build as environment variables.

To https://github.com/jenkinsci/oidc-provider-plugin/pull/16#discussion_r962544835, a Git revision (commit hash) is not currently exposed even for multibranch projects, though it could be I think. (In general a Pipeline build may run multiple checkouts, so there is no such thing as a single definitive version; the `checkout` step’s return value will include the hash for _this_ checkout. The variables defined for multibranch projects apply to the main `checkout scm`.)

I am not really sure what you would use such a claim for anyway, however. Any actual rule in the target system (such as “allow deployments to S3 bucket only for builds associated with the `master` branch”) could not be based on a specific commit, so at most this would be informational.